pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,144 @@ import (
 
 	"k8s.io/kubernetes/pkg/api/latest"
 	"k8s.io/kubernetes/pkg/api/meta"
+	apiutil "k8s.io/kubernetes/pkg/api/util"
+	explatest "k8s.io/kubernetes/pkg/expapi/latest"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
-// Version returns the API version to test against, as set by the KUBE_API_VERSION env var.
-func Version() string {
-	version := os.Getenv("KUBE_API_VERSION")
-	if version == "" {
-		version = latest.Version
+var All = make(map[string]TestGroup)
+var Default TestGroup
+var Experimental TestGroup
+
+type TestGroup struct {
+	Group string
+	// in the format of "group/version"
+	GroupVersionUnderTest string
+}
+
+func init() {
+	kubeTestAPI := os.Getenv("KUBE_TEST_API")
+	if kubeTestAPI != "" {
```

Comment:
We will default the test versions to the latest version of groups several lines below this.

It's better to do the defaulting in the current way. Imaging the env var only contains parts of the groups, e.g.,  KUBE_TEST_API=v1, where the test version of the experimental group is not specified, then we need to default the left groups one by one.